apps/centralised-indexer/src/indexer/RootIndexer.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@nestjs/common'
+import { blockchain as defid } from '@defichain/jellyfish-api-core'
+import { BlockIndexer } from './block/BlockIndexer'
+import { DexSwapIndexer } from './dex-swap/DexSwapIndexer'
+
+export interface Indexer {
+  index: (block: defid.Block<defid.Transaction>) => Promise<void>
+  invalidate: (hash: string) => Promise<void>
+}
+
+/**
+ * Delegates to multiple indexers that are registered here
+ */
+@Injectable()
+export class RootIndexer implements Indexer {
+  private readonly indexers: Indexer[]
+
+  constructor (
+    private readonly blockIndexer: BlockIndexer,
+    private readonly dexSwapIndexer: DexSwapIndexer
+  ) {
+    this.indexers = [
+      blockIndexer,
+      dexSwapIndexer
```

Comment:
> Probably make these 2 have hierarchy instead of same level. `blockIndexer` is story some raw with some some fundamental index. `transactionIndexer` will then derived into `dexSwapIndexer` etc

Sorry, what do you mean by hierarchy?

> Note that `TransactionIndexer.index(transaction)` (instead of block) could be better idea, since we have to process the data transaction by following the `n` chronological order as much as possible.

Ah yes - that's in a separate PR (#1203). Basically to have a `MainTransactionIndexer` similar to Whale?
I didn't include any `Transaction` stuff in this PR to keep it smaller and easier to review 😆